src/Utils/D3D.cpp

```diff
@@ -90,12 +90,17 @@ namespace Util
 	ID3D11DeviceChild* CompileShader(const wchar_t* FilePath, const std::vector<std::pair<const char*, const char*>>& Defines, const char* ProgramType, const char* Program)
 	{
 		auto& device = State::GetSingleton()->device;
-
 		// Build defines (aka convert vector->D3DCONSTANT array)
 		std::vector<D3D_SHADER_MACRO> macros;
+		std::string str = Util::WStringToString(FilePath);
 
-		for (auto& i : Defines)
-			macros.push_back({ i.first, i.second });
+		for (auto& i : Defines) {
+			if (i.first && i.second) {
```

Comment:
Actually, this is just testing they're non-null, it's not testing if `i` is pair. Looks like I need to revert this as it's breaking all shader defines since there are in fact sometimes empty.

Reply:
For example, like a couple lines down:
```
			macros.push_back({ "VR", "" });
```
I believe the empty string may be treated as null.

Reply:
See also https://github.com/doodlum/skyrim-community-shaders/blob/ad396cba237711417b5b45d66dd8ff7b6443050f/src/Deferred.cpp#L612-L616